setup.py

```diff
@@ -155,7 +155,7 @@ def get_git_sha() -> str:
         "dremio": ["sqlalchemy-dremio>=1.1.5, <1.3"],
         "drill": ["sqlalchemy-drill==0.1.dev"],
         "druid": ["pydruid>=0.6.5,<0.7"],
-        "duckdb": ["duckdb-engine==0.9.2"],
+        "duckdb": ["duckdb-engine==0.9.5"],
```

Comment:
As this is a 0.x version, should we restrict it to `<0.10` as per semver breaking change conventions?

Reply:
@villebro for semver shouldn't the upper bound be `<1.0`?

Additionally @guenp providing a bound (as opposed to pinning to an exact version) is preferred as it improves the viable dependency set. It's not atypical for one to have an infeasible dependency set when one explicitly pins a dependency.

Reply:
@john-bodley IIRC, in strict semver, the minor versions of the 0 major are considered breaking. In other words, 0.10 should not be assumed to be backwards compatible with 0.9. @michael-s-molina @eschutho I believe we've discussed this at length at some point, any pointers here would be valuable.

Reply:
Here's the [semver definition](https://semver.org/#spec-item-4) about 0.x:

> 4. Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable.

Reply:
@john-bodley, I agree with you in general, but `"duckdb-engine==0.9.5"` and `"duckdb-engine>=0.9.5, <0.10"` are effectively identical in this case.